From 5ffe7ae722fc0f1d25acbf0257c44d988c1090ce Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Tue, 28 Jan 2020 15:40:34 -0800
Subject: [PATCH 05/13] x86_64-defs.m4: Use X86_GNU_PROPERTY and X86_ENDBR

1. Add X86_ENDBR to PROLOGUE_cpu to place "endbr64" at function entry.
2. Define ASM_END to use X86_GNU_PROPERTY for .note.gnu.property section.

	* mpn/x86_64/x86_64-defs.m4 (PROLOGUE_cpu): Add X86_ENDBR.
	(ASM_END): New.
---
 mpn/x86_64/x86_64-defs.m4 | 4 ++++
 1 file changed, 4 insertions(+)

--- a/mpn/x86_64/x86_64-defs.m4
+++ b/mpn/x86_64/x86_64-defs.m4
@@ -95,6 +95,7 @@ m4_assert_numargs(1)
 	TYPE($1,`function')
 	COFF_TYPE($1)
 $1:
+	X86_ENDBR
 ')
 
 
@@ -167,6 +168,9 @@ ifdef(`PIC',
 	`lea	$1(%rip), $2')
 ')
 
+dnl ASM_END
+
+define(`ASM_END', `X86_GNU_PROPERTY')
 
 define(`DEF_OBJECT',
 m4_assert_numargs_range(2,3)
